src/collective/ifttt/actions/ifttt.py

```diff
@@ -15,41 +16,46 @@
 from zope.schema.vocabulary import SimpleVocabulary
 
 import logging
+import requests
 
 
 logger = logging.getLogger('collective.ifttt')
 
+PAYLOAD_DESCRIPTION = 'description'
+PAYLOAD_USERNAME = 'username'
+PAYLOAD_START = 'start'
+
 payload_options = SimpleVocabulary([
-    SimpleTerm(value=u'Description', title=_(u'Description of action')),
-    SimpleTerm(value=u'Username', title=_(u'Username of Editor')),
-    SimpleTerm(
-        value=u'Event start date/time', title=_(u'Event Start Date/Time')
-    ),
+    SimpleTerm(value=PAYLOAD_DESCRIPTION, title=_(u'Description of action')),
+    SimpleTerm(value=PAYLOAD_USERNAME, title=_(u'Username of Editor')),
+    SimpleTerm(value=PAYLOAD_START, title=_(u'Event Start Date/Time'))
 ])
 
 
 class IIftttTriggerAction(Interface):
     """
         Definition of the configuration available for a  Ifttt action
     """
-    ifttt_name = schema.TextLine(
-        title=_(u'IFTTT event name'),
-        description=_(u'The IFTTT event name'),
-        required=False,
+    Title = schema.TextLine(
+        title=_(u'Title'),
+        description=_(u'Title/Subject for action'),
+        required=True,
+    )
+
+    URL = schema.URI(
```

Comment:
@datakurre @skleinfeldt @tkimnguyen Here is my understanding from above comment, correct me if I'm wrong- 

Basically user has following mandatory inputs while configuring the action:
-  `ifttt_name` >> which will be used to make the post request as `{event}` in  `https://maker.ifttt.com/trigger/{event}/with/key/{secret_key}`?
- 3rd payload selection (`payload_option`)

Ex: if new news item is published:
- `title_of_news_item`: welcome xyz to Plone with a *link* to profile 
- `url_of_news_item`: https://www.plone.com/news/new_joining

For above example `data` in post requests to ifttt will contain - 
 - title: `title_of_news_item`
 - url: `url_of_news_item`
 - payload_option: `payload_option.value()`

Reply:
@Shriyanshagro I believe you have understood correctly. Expect that I'm not sure what you mean with `payload_option.value()`?

So, payload option can be description, username or start. If it's description, the post should contain description of the news item, if it's username, it should contain username of the currently logged-in user (available through plone.api), and if it's start, it might be "effective date" of the news item or null, and only have value if the context is Event (instead of news item), when it would be start value of that event.

Reply:
@Shriyanshagro The sense of your statement about the mandatory inputs is correct and that is also my understanding of the call to `maker.ifttt.com`. I am not following in the code so I can't comment on the example. I would suggest that `ifttt_event_name` would be a clearer variable name than `ifttt_name` (assuming it's a variable!)

Reply:
@datakurre "start" will have "effective date" value only if the current content object (context) has such a field - so Event content type or some other content type with start/end/event behavior?

Reply:
@skleinfeldt nice suggestion for the variable name 👍 
I'll follow it up

Reply:
Re: "'start' will have 'effective date' value", I may misunderstand but I don't think the start date of an Event has anything to do with its effective (publication) date. Those two values will usually be different.

Reply:
For clarity. Let’s let ”start” payload be null when content object does not have start field. (In the default content types only event has start.) I was wrong in suggesting it to have multiple meanings, already causing confusion :)